components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt

```diff
@@ -51,6 +51,15 @@ interface Engine {
         override fun hashCode() = types
     }
 
+    /**
+     * Starts up this engine and makes sure all required initialization logic is executed. The
+     * details are specific to individual implementations, but the following must be true:
+     *
+     * - The engine must be operational after this method was called successfully
+     * - Calling this method on an engine that is already started has no effect
+     */
+    fun start() = Unit
```

Comment:
nit: Following `AbstractCustomTabsService` I think I'd just call it `warmup()`.

Reply:
OK, changed!